### PR TITLE
Geo View

### PR DESCRIPTION
**Pull request by Aelbannan**
Some things to note:

* I changed the App.scss to allow overflow, as allowing the timeline to scroll would have required a large amount of changes otherwise.

This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/ratel/151)

### CODE REVIEW

**Comment by Aelbannan** [comment]
So what's happening is that you need to install go-bindata using

```# You may need to manually install go-bindata using the following command
go get github.com/jteeuwen/go-bindata/go-bindata```

There's an easier way to run ratel though. Go into the client folder, run `npm install` and then `npm start`.

**Comment by campoy** [comment]
Is there any reason why we're not using something like geojson.io? 

Currently, for this query:

```
{
  tourist(func: near(loc, [-122.469829, 37.771935], 10000) ) {
    name
    location: loc
  }
}
```

I get this view:

Screen Shot 2020-01-16 at 6 13 55 PM

While after massaging a bit the query's response, geojson.io shows this:

Screen Shot 2020-01-16 at 6 15 02 PM

For reference, I massaged the data with this jq sequence:

```
cat stuff.json | jq '.data.tourist' | jq 'map({"type": "Feature", "properties": {"name": .name}, "geometry": .location})' | jq '{"type": "FeatureCollection", "features": . }' | pbcopy
```

**Comment by Aelbannan** [comment]
> The build should work with Go modules (but actually the whole thing should be migrated off go-bindata eventually). Probably worth filing an issue about it.

I agree. Go modules were my first thought too, it took me way too long to figure out how to build Ratel. I will file an issue with this.

> This mutation crashes the whole page

This seems to be an existing bug not caused by this PR, I'll take a look at this. Did you have the geo view open?

> Is there any reason why we're not using something like geojson.io?

Honestly, just an oversight from me. I didn't expect people to zoom in that far. Geojson.io is abandoned, but I have found https://leafletjs.com, which actually seems better. I'll integrate that into the geo view.

Thanks for the thorough review, I really do appreciate it!

**Comment by Aelbannan** [comment]
Maps are now vastly updated. 

image

It should now function similarly to geojson.io. Also, the near function now expands in size depending on the distance argument.

Let me know if you have any other feedback.

**Comment by Aelbannan** [comment]
> Lets launch these new views one by one in separate PRs. This way PRs / reviews will be smaller, issues will get caught faster, etc.

Sure. I'll keep this branch for the geo view.

> instead of always present on screen can these buttons only show up when geo data is detected?

I can, I just want people to know they exist. I'll see if I can add some sort of guide in the editor that explains viewing options.

**Comment by Aelbannan** [comment]
> Code in index.html is like kung-gu - we should never add it unless it absolutely cannot be avoided

Makes sense, will either include the js file or find a npm package.



> > fullscreen: isFullscreen,
                collapsed,
                "h-100": true,


> What is this magic number and can its CSS be instead incorporated into GeoView.scss?


It's a bootstrap class: https://getbootstrap.com/docs/4.4/utilities/sizing/

Pretty much it just sets height: 100%. If you don't like these small classes I can remove them, I mostly use them for clarity.

> other files do export default function ComponentName(...) {...} this sometimes helps dev tools tell where the rendered fragment came from (or at least used **to)**

Good point. Will do.

> Please wrap all CSS into a component-specific selector, my understanding is current implementation is modifying classes .options-button and .error-alert globally

Sure, that's better practice.

File: client/src/components/ConsolePage/GeoView.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2019 Dgraph Labs, Inc. and Contributors
+//
+// Licensed under the Dgraph Community License (the "License"); you
+// may not use this file except in compliance with the License. You
+// may obtain a copy of the License at
+//
+//     https://github.com/dgraph-io/ratel/blob/master/LICENSE
+
+@import url("https://unpkg.com/leaflet@1.6.0/dist/leaflet.css");
```
Comment (paulftw) [inline client/src/components/ConsolePage/GeoView.scss]: Please import it from the node_modules, not from unpkg, I've pasted an example of doing it above.
Import from CDN won't work offline, and we have (semi?)-official support of ratel on admin machines w/o internet access.

**Pull request merged**